cli/Sources/TuistGenerator/Generator/ProjectDescriptorGenerator.swift

```diff
@@ -306,6 +311,31 @@ final class ProjectDescriptorGenerator: ProjectDescriptorGenerating {
         pbxProject.localPackages = localPackageReferences.sorted { $0.key < $1.key }.map { $1 }
     }
 
+    private func addLocalPackageReference(
+        root: PBXGroup,
+        groupPath: String,
+        reference: PBXFileReference
+    ) throws {
+        let packagesGroup: PBXGroup?
+        if let existing = root.group(named: "Packages") {
+            packagesGroup = existing
+        } else {
+            packagesGroup = try root.addGroup(named: "Packages", options: .withoutFolder).first
+        }
+
+        let components = groupPath.split(separator: "/").map(String.init)
+        var currentGroup = packagesGroup
+        for folder in components {
+            if let sub = currentGroup?.group(named: folder) {
+                currentGroup = sub
+            } else {
+                currentGroup = try currentGroup?.addGroup(named: folder, options: .withoutFolder).first
+            }
+        }
+
+        currentGroup?.children.append(reference)
```

Comment:
What about calling [this API instead](https://github.com/tuist/XcodeProj/blob/eea1743e4949459026a86838ece3d19a5e956392/Sources/XcodeProj/Objects/Files/PBXGroup.swift#L146) which already addresses the creation of intermediate groups?